packages/kbn-analytics/src/reporter.ts

```diff
@@ -63,14 +64,14 @@ export class Reporter {
     this.storageManager.store(this.reportManager.report);
   }
 
-  public start() {
+  public start = () => {
     if (!this.interval) {
       this.interval = setTimeout(() => {
         this.interval = null;
         this.sendReports();
       }, this.checkInterval);
     }
-  }
+  };
```

Comment:
I use classes as functional `namespaces`. Using lambdas binds `this` which is nice; it also makes it harder to override methods since you cannot call `super.method` in the overriding method; which gets rid of many of the inheritance issues. I've converted everythint to lambda in this class simply to avoid this binding issues. As i've seen `if (this)` was added to one of the methods since I have no control over what `this` would bind to those public methods.

Reply:
Yeah, I get it... don't sweat it, the change seems reasonable, I just dislike the fact that we have to do it.
It's why I . tend to stay away from Classes if I can 🤷‍♂️

Definitely not a blocker. :)